climada/engine/unsequa/test/test_unsequa.py

```diff
@@ -527,92 +571,134 @@ def test_calc_uncertainty_pool_pass(self):
+               'test_si_name' : ['CV', 16],
+               'test_si_value' : [0.25000, 2]
+           },
+           'hdmr': {
+               'sampling_method' : 'saltelli',
+               'sampling_kwargs' : {},
+               'N' : 100,
+               'sensitivity_kwargs' : {},
+               'test_param_name' : ['x_exp', 2],
+               'test_si_name' : ['Sa', 4],
+               'test_si_value' : [0.004658, 3]
+           },
+           'ff': {
+
+               'sampling_method' : 'ff',
+               'sampling_kwargs' : {'seed' : 12345},
+               'N' : 4,
+               'sensitivity_kwargs' : {'second_order': True},
+               'test_param_name' : ['x_exp', 0],
+               'test_si_name' : ['IE', 4],
+               'test_si_value' : [865181825.901295, 10]
+           },
+           'sobol': {
+               'sampling_method' : 'saltelli',
+               'sampling_kwargs' : {},
+               'N' : 4,
+               'sensitivity_kwargs' : {},
+               'test_param_name' : ['x_paa', 5],
+               'test_si_name' : ['ST', 8],
+               'test_si_value' : [0.313025, 10]
+           },
+
+            'dgsm': {
+                'sampling_method' : 'finite_diff',
+                'N' : 4,
+                'sampling_kwargs' : {'seed':12345},
+                'sensitivity_kwargs' : {'num_resamples': 100,
+                                        'conf_level': 0.95, 'seed': 12345},
+                'test_param_name' : ['x_exp',0],
+                'test_si_name' : ['dgsm', 8],
+                'test_si_value' : [1.697516e-01, 9]
+            },
+            'fast': {
+                'sampling_method' : 'fast_sampler',
+                'sampling_kwargs' : {'M' : 4, 'seed' : 12345},
+                'N' : 256,
+                'sensitivity_kwargs' : {'M': 4, 'seed': 12345},
+                'test_param_name' : ['x_exp',0],
+                'test_si_name' : ['S1_conf',8],
+                'test_si_value' : [0.671396, 1]
+            },
+
+            'rbd_fast': {
+                'sampling_method' : 'saltelli',
+                'sampling_kwargs' : {},
+                'N' : 24,
+                'sensitivity_kwargs' : {'M': 4, 'seed': 12345},
+                'test_param_name' : ['x_exp', 0],
+                'test_si_name' : ['S1_conf', 4],
+                'test_si_value' : [0.152609, 4]
+            },
+
+            'morris': {
+                'sampling_method' : 'morris',
+                'sampling_kwargs' : {'seed': 12345},
+                'N' : 4,
+                'sensitivity_kwargs' : {},
+                'test_param_name' : ['x_exp', 0],
+                'test_si_name' : ['mu', 1],
+                'test_si_value' : [5066460029.63911, 8]
+            },
+        }
+
+        def test_sensitivity_method(sensitivity_method, param_dict):
+            """Function to test each seaprate sensitivity method"""
+            unc_calc = CalcImpact(exp_unc, impf_unc, haz_unc)
```

Comment:
It would be cleaner if the exp_unc, impf_unc and haz_unc are also passed as arguments.